src/zenml/utils/artifact_utils.py

```diff
@@ -427,6 +433,30 @@ def upload_artifact(
     return response.id
 
 
+def _get_new_artifact_version(artifact_name: str) -> int:
+    """Get the next auto-incremented version for an artifact name.
+
+    Args:
+        artifact_name: The name of the artifact.
+
+    Returns:
+        The next auto-incremented version.
```

Comment:
Is it too obvious to mention that auto-incremented means +1?

Reply:
Because people can have non-int versions, too, no?

Reply:
+1 to this question

Reply:
Generally, how these auto-versions will cross with custom versions? Isn't it too confusing for users? Maybe going with Version Name / Version Number approach as we have in Model Versions?

Reply:
> Is it too obvious to mention that auto-incremented means +1?

In my opinion, yes, since it's a super short function that's not even user-facing.

In the docs I'd explain that in more detail of course

Reply:
> > Generally, how these auto-versions will cross with custom versions? Isn't it too confusing for users? Maybe going with Version Name / Version Number approach as we have in Model Versions?
> 
> No, I still think that version name + number is a fundamentally flawed design from a UX perspective

I think you came to this already, but I was not saying that you need to ask users for numbers and names, but have it in DB, same as we do now in Models - user can input 1, "1", "Some name", ModelStages.STAGING, "staging", etc all in the same field. The same logic can be achieved here.
Initially, I was also against extracting numbers, but it is simplifying backend life a bit, IMO. The only downside is that you have to ban numerical version names from being inputted by the user, if you rely on the uniqueness of a name and auto-populate the name with a number, if no user input is given.

Reply:
Yeah that's pretty much what I did now:
- in the models and all user-facing methods, `version` is a single field that is `Union[str, int]`, which is intuitive and also allows comparisons of int versions as one would expect.
- in the DB I added version_number as an optional field which is only set for numbered versions to make it more efficient to retrieve numbered versions only (e.g. for auto-increment)

Reply:
At the moment I don't need to ban numeric custom versions and I don't see yet why I would need to - anyway, I will implement that in the follow-up PR that adds manual string versions if we need it